File: A000217/A000217.py
# Triangular numbers: a(n) = binomial(n+1,2) = n(n+1)/2 = 0 + 1 + 2 + ... + n.
# http://oeis.org/A000217

import math

# Algorithm given by sum of arithmetic progression
print("n*(n+1)/2 algorithm")
def T(n):
    return int(n*(n+1)/2)
for i in range(30):
    print(T(i), end=" ")
print();

# Algorithm given by the binomial(n+1, 2)
print("binomial(n+1, 2)")
def binomial_coefficient(n, k):
    if n < k:
        return 0
    else:
        return int(math.factorial(n)/(math.factorial(k)*math.factorial(n - k)))
for i in range(0, 30):
    print(binomial_coefficient(i+1, 2), end=" ")
print()

File: A000217/A000217.c
// Triangular numbers: a(n) = binomial(n+1,2) = n(n+1)/2 = 0 + 1 + 2 + ... + n.
// http://oeis.org/A000217

#include <stdio.h>

// Algorithm given by the sum of the arithmetic progression 1 + 2 + 3 + ... + n
int T(int n)
{
    return (n*(n+1))/2;
}

int main(void)
{
    printf("n*(n+1)/2 algorithm\n");
    for (int i = 0; i < 30; i++)
    {
        printf("%d ", T(i));
    }
    printf("\n");
}

